libraries/AP_NavEKF/AP_NavEKF_Source.h

```diff
@@ -60,35 +62,48 @@ class AP_NavEKF_Source
     // initialisation
     void init();
 
+    uint8_t getActiveSourceSet(uint8_t core_number) const {
+        // check if we are using a separate source set for each core
+        if (option_is_set(SourceOptions::SRC_PER_CORE)) {
+            if (core_number >= AP_NAKEKF_SOURCE_SET_MAX) {
+                // this is a bit of a hack, but we need to return a valid source set
```

Comment:
maybe remove the hack, just say what we're doing which is "invalid core index, return active source set"